core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -74,6 +74,7 @@ object Defaults {
   val InitialBrokerRegistrationTimeoutMs = 60000
   val BrokerHeartbeatIntervalMs = 2000
   val BrokerSessionTimeoutMs = 9000
+  val MetadataSnapshotMinNewRecordBytes = Int.MaxValue
```

Comment:
Created this Jira to fix this before 3.0 release: https://issues.apache.org/jira/browse/KAFKA-12974

Reply:
Hmm... why not just set this to 100 MB for now (or whatever default we consider reasonable?)

It won't harm anything to generate snapshots that the broker won't use, right? The controller itself can use them to start up more quickly.

Reply:
Until we fix https://issues.apache.org/jira/browse/KAFKA-12155, the current implementation may delete log segments. This will force new brokers to fetch a snapshot which they don't how to load until we fix https://issues.apache.org/jira/browse/KAFKA-12647

Reply:
OK